crates/bevy_reflect/bevy_reflect_derive/src/impls/structs.rs

```diff
@@ -0,0 +1,205 @@
+use crate::impls::impl_typed;
+use crate::ReflectStruct;
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{Index, Member};
+
+/// Implements `Struct`, `GetTypeRegistration`, and `Reflect` for the given derive data.
+pub(crate) fn impl_struct(reflect_struct: &ReflectStruct) -> TokenStream {
```

Comment:
This is the main change to these impl modules: using `ReflectStruct` in place of `ReflectDeriveData` (or similar struct).